ee/api/billing.py

```diff
@@ -87,46 +87,49 @@ def patch(self, request: Request, *args: Any, **kwargs: Any) -> Response:
 
         return self.list(request, *args, **kwargs)
 
+    class ActivationSerializer(serializers.Serializer):
+        products = serializers.CharField(required=True)
+        redirect_path = serializers.CharField(required=False)
+
     @action(methods=["GET"], detail=False)
     def activation(self, request: Request, *args: Any, **kwargs: Any) -> HttpResponse:
         license = get_cached_instance_license()
         organization = self._get_org_required()
 
-        redirect_path = request.GET.get("redirect_path") or "organization/billing"
+        serializer = self.ActivationSerializer(data=request.GET)
+        serializer.is_valid(raise_exception=True)
+
+        redirect_path = serializer.validated_data.get("redirect_path", "organization/billing")
         if redirect_path.startswith("/"):
             redirect_path = redirect_path[1:]
 
         redirect_uri = f"{settings.SITE_URL or request.headers.get('Host')}/{redirect_path}"
         url = f"{BILLING_SERVICE_URL}/activation?redirect_uri={redirect_uri}&organization_name={organization.name}"
 
-        plan = request.GET.get("plan", None)
-        product_keys = request.GET.get("products", None)
-        if not plan and not product_keys:
```

Comment:
`plan` is no longer supported - can we remove it entirely, or should we keep it with a custom error?

Reply:
I feel like I keep running into situations where I'm confident something is fully removed and get proven wrong. I'd say leave it in with an error so we can track in sentry where the stragglers are.